src/main/java/com/spikes2212/command/genericsubsystem/commands/smartmotorcontrollersubsystem/MoveSmartMotorControllerSubsystem.java

```diff
@@ -0,0 +1,153 @@
+package com.spikes2212.command.genericsubsystem.commands.smartmotorcontrollersubsystem;
+
+import com.ctre.phoenix.motorcontrol.ControlMode;
+import com.revrobotics.CANSparkMax;
+import com.spikes2212.command.genericsubsystem.smartmotorcontrollersubsystem.SmartMotorControllerSubsystem;
+import com.spikes2212.control.FeedForwardSettings;
+import com.spikes2212.control.PIDSettings;
+import edu.wpi.first.wpilibj.Timer;
+import edu.wpi.first.wpilibj2.command.CommandBase;
+
+import java.util.function.Supplier;
+
+/**
+ * A command that moves a {@link SmartMotorControllerSubsystem} using its master motor controller's control loops.
+ *
+ * @author Yoel Perman Brilliant
+ * @see SmartMotorControllerSubsystem
+ */
+public class MoveSmartMotorControllerSubsystem extends CommandBase {
+
+    /**
+     * The {@link SmartMotorControllerSubsystem} this command will run on.
+     */
+    protected final SmartMotorControllerSubsystem subsystem;
+
+    /**
+     * The setpoint this command should bring the {@link SmartMotorControllerSubsystem} to.
+     */
+    protected final Supplier<Double> setpoint;
+
+    /**
+     * The loop's control mode (e.g. voltage, velocity, position...). Only applicable when
+     * using a CTRE motor controller.
+     */
+    protected final ControlMode controlMode;
+
+    /**
+     * The loop's control type (e.g. voltage, velocity, position...). Only applicable when
+     * using a {@link CANSparkMax} motor controller.
+     */
+    protected final CANSparkMax.ControlType controlType;
+
+    /**
+     * The loop's pid constants.
+     */
+    protected final PIDSettings pidSettings;
+
+    /**
+     * The loop's feed forward gains.
+     */
+    protected final FeedForwardSettings feedForwardSettings;
+
+    /**
+     * The most recent timestamp on which the loop has not reached its target setpoint.
+     */
+    private double lastTimeNotOnTarget;
+
+    /**
+     * Constructs a new (generic) instance of {@link MoveSmartMotorControllerSubsystem}.
+     *
+     * @param subsystem the {@link SmartMotorControllerSubsystem} this command will run on
+     * @param pidSettings the loop's pid constants
+     * @param feedForwardSettings the loop's feed forward gains
+     * @param controlMode the loop's control mode (e.g. voltage, velocity, position...).
+     *                    Only applicable when using CTRE motor controllers
+     * @param controlType the loop's control mode (e.g. voltage, velocity, position...).
+     *                    Only applicable when using {@link CANSparkMax} motor controllers
+     * @param setpoint the setpoint this command should bring the {@link SmartMotorControllerSubsystem} to
+     */
+    protected MoveSmartMotorControllerSubsystem(SmartMotorControllerSubsystem subsystem, PIDSettings pidSettings,
```

Comment:
should this be protected? know that a change of this constructor affects the other constructors as well because they use it.

Reply:
Yes, it probably shouldn't be public since it should never realistically be used (since it allows both talon and spark max control mode) but it's useful for easy extention.